src/lib.rs

```diff
@@ -183,30 +181,30 @@
 //!
 //! At last we create our `main.rs` file just like we would have with the previous two examples:
 //!
-//! ```ignore
+//! ```no_run
+//! # #[cfg(feature = "yaml")]
+//! # fn main() {
+//! # use clap::{App, load_yaml};
 //! // (Full example with detailed comments in examples/17_yaml.rs)
 //! //
 //! // This example demonstrates clap's building from YAML style of creating arguments which is far
 //! // more clean, but takes a very small performance hit compared to the other two methods.
-//! #[macro_use]
-//! extern crate clap;
-//! use clap::App;
-//!
-//! fn main() {
-//!     // The YAML file is found relative to the current file, similar to how modules are found
-//!     let yaml = load_yaml!("cli.yml");
-//!     let matches = App::from_yaml(yaml).get_matches();
-//!
-//!     // Same as previous examples...
-//! }
+//! //
+//! // The YAML file is found relative to the current file, similar to how modules are found
+//! let yaml = load_yaml!("cli.yml");
```

Comment:
This and the next line fail because for some reason they're being compiled without the `yaml` feature included. Not entirely sure why this is... Might be interesting to run `cargo expand` on it with and without `yaml` enabled